src/DataProtection/DataProtection/src/KeyManagement/KeyRingBasedDataProtector.cs

```diff
@@ -318,34 +300,18 @@ private byte[] UnprotectCore(byte[] protectedData, bool allowOperationsOnRevoked
             }
         }
 
-        // Helper function to write a GUID to a 32-bit alignment; useful on ARM where unaligned reads
-        // can result in weird behaviors at runtime.
-        private static void Write32bitAlignedGuid(void* ptr, Guid value)
+        private static void WriteGuid(void* ptr, Guid value)
         {
-            Debug.Assert((long)ptr % 4 == 0);
-
-            if (BitConverter.IsLittleEndian)
-            {
-                ((int*)ptr)[0] = ((int*)&value)[0];
-                ((int*)ptr)[1] = ((int*)&value)[1];
-                ((int*)ptr)[2] = ((int*)&value)[2];
-                ((int*)ptr)[3] = ((int*)&value)[3];
-            }
-            else
-            {
-                // The first 8 bytes of the Guid hold one 32-bit integer
-                // and two 16-bit integers that must be byte-swapped.
-                ((byte*)ptr)[0] = ((byte*)&value)[3];
-                ((byte*)ptr)[1] = ((byte*)&value)[2];
-                ((byte*)ptr)[2] = ((byte*)&value)[1];
-                ((byte*)ptr)[3] = ((byte*)&value)[0];
-                ((byte*)ptr)[4] = ((byte*)&value)[5];
-                ((byte*)ptr)[5] = ((byte*)&value)[4];
-                ((byte*)ptr)[6] = ((byte*)&value)[7];
-                ((byte*)ptr)[7] = ((byte*)&value)[6];
-                ((int*)ptr)[2] = ((int*)&value)[2];
-                ((int*)ptr)[3] = ((int*)&value)[3];
-            }
+#if NETCOREAPP
+            var span = new Span<byte>(ptr, sizeof(Guid)); // performs appropriate endianness fixups
```

Comment:
For clarity, I'd put the comment on the following line, not on this line. It's the `Guid.TryWriteBytes` method that performs fixups, not the `Span<T>` ctor.

Reply:
oh right, I copied that from the Read one and forgot to move it :smile: